### ISSUE TITLE
PartDesign: Support ability to create holes from sketch points (instead of the extra step of creating a circle)

### ISSUE THREAD

**Issue by wwmayer** [FreeCAD/FreeCAD#6884]
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?f=8&t=68696

### Subproject(s) affected?

PartDesign

### Idea description

Currently a circle must be added to a sketch to allow to make a hole afterwards. Since the radius of a circle is ignored anyway it would be convenient to just add points and use them for the holes.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by zackwhit** [comment]
I've been working on this a bit, gotten a draft of the feature working. I will update my fork with the changes once I can refine them a bit. Nobody else has already fixed this, right? lol

![image](https://user-images.githubusercontent.com/16970830/188124009-5e5e22b6-af4e-4193-9b21-cd9c1f421549.png)

[learning notes as I go](https://www.zackwhitlock.com/post/freecad-devlog-2-5-points-as-a-basis-for-holes)

**Comment by pierreporte** [comment]
I commented in https://github.com/FreeCAD/FreeCAD/issues/10100#issuecomment-1783929916 that it could be useful to have two kinds of points for construction geometry and regular geometry. It would allow to build complex patterns with a lot of geometry and only make visible from outside the sketch the position of holes using regular points.

**Comment by pierreporte** [comment]
I opened an issue (#11911) to implement normal points in sketches (currently there are only construction points). They are a requirement to implement this feature request.

**Comment by maxwxyz** [comment]
Remarks from the FreeCAD Day 2024 Complaint Session:
I've got an unconstrained sketch and then the need to draw circles of arbitrary sizes.
We need the sketch to say where your holes are going to go, but you shouldn't need to draw a circle. You should have something like a point or location/layout.
You want to say there are features in these five places. And then you go into the hole tool and it will put the features in those places. And currently you just draw a circle of arbitrary. So you only draw the circle to have the point in the center.

**Comment by maxwxyz** [comment]
@zackwhit are you still working on this?

**Comment by zackwhit** [comment]
> @zackwhit are you still working on this?

No. I put it all on pause while waiting for the TNP merge; no idea where it stands now.

**Comment by maxwxyz** [comment]
@zackwhit the TNP mitigation code has been merged and is in the stable 1.0 version. Are you planning to pick this up again?

**Comment by zackwhit** [comment]
> @zackwhit the TNP mitigation code has been merged and is in the stable 1.0 version. Are you planning to pick this up again?

Not planning on picking it up again; at least anytime soon.

**Comment by theo-vt** [comment]
Hey! this has been bugging me as well and I thought I would give it a try,

I was wondering if it would be doable to add a point type to the sketcher to differentiate between a "curve handle" and a "point"

![Image](https://github.com/user-attachments/assets/3240bfa1-3198-4481-9f0c-1e99c4777e48)

Inventor has little spikes on points explicitly added by the user [left] by opposition to curve handles [right] and I think it looks neat. The hole tool could target all explicit points by default in addition to circles/arcs and the selected points otherwise?

thanks,

**Comment by pierreporte** [comment]
@theo-vt There is already a dedicated point type (the point tool in sketcher), a proposal to use it better than now (see https://github.com/FreeCAD/FreeCAD/issues/13391#issuecomment-2585909362), and a proposal for how everything should look (in the design working group space on the FreeCAD Discord server).

**Comment by theo-vt** [comment]
Thanks for the pointers, I'll look into that!

**Comment by theo-vt** [comment]
I am not sure how to approach the problem and am wondering if it would make sense to modify the profile before or as soon as it gets to the Hole class such that points are transformed into circles if no circle/arc is found in the profile? After that the normal hole code could be kept exactly the same,

**Comment by pierreporte** [comment]
The problem with this approach is that there will not be any error or warning if no defining point is found in the sketch. IMO the best way is to check the version of the document file, and act differently if it's older that the version the feather was merged.

**Comment by pierreporte** [comment]
IIRC @kadet1090 has experience with migrations.

**Comment by K4LCIFER** [comment]
What release was this feature added in? I don't see it in FreeCAD v1.0.2, but given that this was merged in April, I feel like it should be there.
